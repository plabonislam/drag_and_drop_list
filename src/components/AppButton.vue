<template>
  <button :class="{ loading: isLoading }" @click="handleClick" :disabled="isLoading">
    <span v-if="isLoading">Loading...</span>
    <span v-else>{{ label }}</span>
  </button>
</template>

<script setup>
import { ref, defineEmits, defineProps } from 'vue'

const props = defineProps({
  label: {
    type: String,
    default: 'Click me',
    required: true
  }
})

const emit = defineEmits(['loaded'])
const isLoading = ref(false)

const handleClick = () => {
  isLoading.value = true
  setTimeout(() => {
    isLoading.value = false
    emit('loaded')
  }, 2000)
}
</script>

<style scoped>
.loading {
  background-color: gray;
}
</style>
