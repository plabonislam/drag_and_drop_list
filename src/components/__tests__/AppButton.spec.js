import { describe,it ,expect } from "vitest";
import { mount} from "@vue/test-utils";
import AppButton from "../AppButton.vue";


describe('APPButton',()=>{

    it("renders component",()=>{
        const wrapper = mount(AppButton,{
            props:{
                text:"click me"
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
       
    })
})