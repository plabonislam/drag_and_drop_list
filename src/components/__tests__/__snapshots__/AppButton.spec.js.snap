// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`APPButton > renders component 1`] = `"<button data-v-19765ba3="" class="" text="click me"><span data-v-19765ba3="">Click me</span></button>"`;
